pkg/graphql/fields.go

```diff
@@ -7,16 +7,30 @@ import (
 )
 
 type RestrictedFieldsValidator interface {
-	Validate(operation *ast.Document, restrictedFields []fields.Type) (RestrictedFieldsResult, error)
+	Validate(operation, definition *ast.Document, restrictedTypes []fields.Type) (RestrictedFieldsResult, error)
 }
 
 type fieldsValidator struct {
 }
 
-func (d fieldsValidator) Validate(operation *ast.Document, restrictedFields []fields.Type) (RestrictedFieldsResult, error) {
+func (d fieldsValidator) Validate(operation, definition *ast.Document, restrictedTypes []fields.Type) (RestrictedFieldsResult, error) {
 	report := operationreport.Report{}
+	if len(restrictedTypes) == 0 {
+		return restrictedFieldsResult(true, report)
+	}
 
-	// call fields visitor
+	requestTypes := make(fields.RequestTypes)
+	fields.NewGenerator().Generate(operation, definition, &report, requestTypes)
+
+	for i := 0; i < len(restrictedTypes); i++ {
```

Comment:
Instead of writing this with indexes, wouldn't it be better to right with value iteration like:
`for _,restrictedType:=range restrictedTypes {`

it makes code more readable